server/services/zwavejs-ui/utils/convertToGladysDevice.js

```diff
@@ -0,0 +1,59 @@
+const get = require('get-value');
+
+const { EXPOSES } = require('../lib/constants');
+
+const cleanNames = (text) => {
+  if (!text || typeof text !== 'string') {
+    return '';
+  }
+  return text
+    .replaceAll(' ', '_')
+    .replaceAll('(', '')
+    .replaceAll(')', '')
+    .toLowerCase();
+};
+
+const getDeviceFeatureExternalId = (location, name, endpoint, comClass, property, propertyKey) =>
+  `zwavejs-ui:${location}:${name}:${endpoint}:${comClass}:${property}:${propertyKey}`;
+
+const convertToGladysDevice = (serviceId, device) => {
+  const features = [];
+
+  // Foreach value, we check if there is a matching feature in Gladys
+  Object.keys(device.values).forEach((valueKey) => {
+    const value = device.values[valueKey];
+    const { commandClassName, property, propertyKey, endpoint } = value;
+    const commandClassNameClean = cleanNames(commandClassName);
+    const propertyClean = cleanNames(property);
+    const propertyKeyClean = cleanNames(propertyKey);
+    const exposeFound = get(EXPOSES, `${commandClassNameClean}.${propertyClean}.${propertyKeyClean}`);
+    if (exposeFound) {
+      features.push({
+        ...exposeFound,
+        name: value.id,
+        external_id: getDeviceFeatureExternalId(
+          device.loc,
+          device.name,
+          endpoint,
+          commandClassNameClean,
+          propertyClean,
+          propertyKeyClean,
+        ),
+      });
+    }
+  });
+
+  return {
+    name: device.name,
+    external_id: `zwavejs-ui:${device.id}`,
```

Comment:
Maybe we should keep consistency with feature ID, in case we implement a "poll" or someting to realign features... so we'll have the mqtt topic in the device too.

`zwavejs-ui:${location}:${name}`

Reply:
Good idea!